.github/workflows/bvt-clang.yml

```diff
@@ -7,32 +7,41 @@ jobs:
     steps:
     - uses: actions/checkout@v4
 
+    - name: install meson
+      run: pipx install meson
+
     - name: install clang
       run: |
-        wget https://apt.llvm.org/llvm.sh
-        chmod +x llvm.sh
-        sudo ./llvm.sh 21
-        sudo apt install -y libc++-21-dev clang-format-21
+        curl https://apt.llvm.org/llvm.sh | sudo bash -s -- 21
+        sudo apt install -y clang-21 libc++-21-dev clang-format-21
+        cat <<'EOF' >> "$GITHUB_ENV"
+        CC=clang-21
+        CXX=clang++-21
+        CXXFLAGS=-stdlib=libc++
+        EOF
 
     - name: check compiler version
       run: |
-        clang++-21 --version
+        "$CXX" --version
 
-    - name: build and run test with clang 21
+    - name: build and run test with clang 20 on cmake
       run: |
-        cmake -B build -GNinja -DCMAKE_C_COMPILER=clang-21 -DCMAKE_CXX_COMPILER=clang++-21 -DCMAKE_CXX_FLAGS="-stdlib=libc++" -DCMAKE_CXX_STANDARD=23 -DCMAKE_BUILD_TYPE=Release -DPROXY_BUILD_MODULES=TRUE
+        cmake -B build -GNinja -DCMAKE_CXX_STANDARD=23 -DCMAKE_BUILD_TYPE=Release -DPROXY_BUILD_MODULES=TRUE
         mapfile -t FILES < <(find include tests benchmarks build/examples_from_docs tools -type f \( -name '*.h' -o -name '*.ixx' -o -name '*.cpp' \))
         echo "Running clang-format on ${#FILES[@]} files: ${FILES[*]}"
         clang-format-21 --dry-run --Werror "${FILES[@]}"
         cmake --build build -j
         ctest --test-dir build -j
         mkdir build/drop
-        chmod +x tools/dump_build_env.sh
-        ./tools/dump_build_env.sh clang++-21 build/drop/env-info.json
+        bash ./tools/dump_build_env.sh "$CXX" build/drop/env-info.json
 
-    - name: run benchmarks
+    - name: build and run test with clang 20 on meson
       run: |
-        build/benchmarks/msft_proxy_benchmarks --benchmark_min_warmup_time=0.1 --benchmark_min_time=0.1s --benchmark_repetitions=30 --benchmark_enable_random_interleaving=true --benchmark_report_aggregates_only=true --benchmark_format=json > build/drop/benchmarking-results.json
+        meson setup builddir --buildtype=release -Dtests=enabled
```

Comment:
CI logs indicate modules tests weren't built under Meson. We should ensure Meson options enable modules tests equivalently to CMake.